File: tests/Generator/resource/java/complex/ArrayDefinitionType.java
package org.typeapi.model;

import com.fasterxml.jackson.annotation.*;

@JsonClassDescription("Represents an array which contains a dynamic list of values of the same type")
public class ArrayDefinitionType extends CollectionDefinitionType {
    private String type = "array";

    @JsonSetter("type")
    public void setType(String type) {
        this.type = type;
    }

    @JsonGetter("type")
    public String getType() {
        return this.type;
    }
}

File: tests/Generator/resource/php/complex/ArrayDefinitionType.php
<?php

declare(strict_types = 1);

namespace TypeAPI\Model;

use PSX\Schema\Attribute\Description;

#[Description('Represents an array which contains a dynamic list of values of the same type')]
class ArrayDefinitionType extends CollectionDefinitionType implements \JsonSerializable, \PSX\Record\RecordableInterface
{
}

